code/edgeidentity/src/main/java/com/adobe/marketing/mobile/edge/identity/IdentityState.java

```diff
@@ -69,15 +78,11 @@ boolean bootupIfReady(final SharedStateCallback callback) {
 			return true;
 		}
 
-		// Load properties from local storage
-		identityProperties = IdentityStorageService.loadPropertiesFromPersistence();
-
-		if (identityProperties == null) {
-			identityProperties = new IdentityProperties();
-		}
-
 		// Reuse the ECID from Identity Direct (if registered) or generate new ECID on first launch
 		if (identityProperties.getECID() == null) {
+			// Wait for all extensions to be registered as forth coming logic depends on Identity Direct state
+			if (!areAllExtensionsRegistered(callback)) return false;
```

Comment:
I think it would be safe to add this to iOS as well. I don't remember the full conversation but I was under the impression that the EventHub shared state was the first (or one of the first) events dispatched when the SDK starts. Adding a check like this for the EventHub shared state makes the code more robust, so it's a good idea.